api/src/opentrons/hardware_control/modules/magdeck.py

```
@@ -114,13 +114,24 @@ async def calibrate(self) -> None:
         await self._driver.probe_plate()
         # return if successful or not?
 
-    async def engage(self, height: float) -> None:
+    # TODO(mc, 2022-09-23): refactor this method to take real mm,
+    # hardware API should abstract away the idea of "short millimeters"
+    # https://opentrons.atlassian.net/browse/RET-1242
+    async def engage(
+        self,
+        height: Optional[float] = None,
+        height_from_base: Optional[float] = None,
+    ) -> None:
         """Move the magnet to a specific height, measured from home position.
 
         The units of position depend on the module model.
         For GEN1, it's half millimeters ("short millimeters").
         For GEN2, it's millimeters.
         """
+        if height is None:
+            assert height_from_base is not None, "An engage height must be specified"
```

Can we raise an error rather than have an assert statement? Also seems like the check should just be `if height is None or height_from_base is None`

---

The check would be `if height is None and height_from_base is None`, not `or`. That being said, I softly prefer `assert` for "this is how you use this internal API." I think raising something else adds boilerplate without value, because if this error is raised then _our_ code is wrong

---

Fine to leave it as is.